### ISSUE TITLE
ibm_dns_custom_resolver id is not the id of the resource is is id:some_extra_string

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#2966]
```
resource "ibm_dns_custom_resolver" "location" {
```

Then later using the custom resolver "id" for the ibm_dns_custom_resolver_forwarding_rule.resolver_id produced the following.  Notice the colon (:) separated string where the first part is the id:

```
  # ibm_dns_custom_resolver_forwarding_rule.location["enterprise"] will be created
  + resource "ibm_dns_custom_resolver_forwarding_rule" "location" {
      + forward_to  = [
          + "192.168.16.9",
          + "192.168.17.9",
        ]
      + id          = (known after apply)
      + instance_id = "3874a29a-ed73-429a-bc02-f2534996f45f"
      + match       = "tvpc-transit.com"
      + resolver_id = "2e6e328e-5e55-4b2e-9a2e-fb5b5b976436:3874a29a-ed73-429a-bc02-f2534996f45f"
      + rule_id     = (known after apply)
      + type        = "zone"
    }

```

It is possible to work around this:

```
resource "ibm_dns_custom_resolver_forwarding_rule" "location" {
  resolver_id = split(":", ibm_dns_custom_resolver.foo.id)[0]
}
```

**Comment by DwitikrDash** [comment]
It's possible. working on it

**Comment by DwitikrDash** [comment]
Hi,

Could you let us know how to reproduce the above scenario (resolver_id = "2e6e328e-5e55-4b2e-9a2e-fb5b5b976436:3874a29a-ed73-429a-bc02-f2534996f45f"). We tried terraform plan but did not see any extra string in resolver id

```
dwitikrushnadash@Dwitikrushnas-MacBook-Pro custom-resolver % terraform plan
ibm_dns_custom_resolver.test: Refreshing state... [id=5191162b-5c45-4e96-851f-ebadc80a7e8f:7cea4fae-0440-4164-ac65-de0520ae8d90]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # ibm_dns_custom_resolver_forwarding_rule.dns_custom_resolver_forwarding_rule will be created
  + resource "ibm_dns_custom_resolver_forwarding_rule" "dns_custom_resolver_forwarding_rule" {
      + description = "test forward rule"
      + forward_to  = [
          + "168.20.22.122",
        ]
      + id          = (known after apply)
      + instance_id = "7cea4fae-0440-4164-ac65-de0520ae8d90"
      + match       = "test.example.com"
      + resolver_id = "5191162b-5c45-4e96-851f-ebadc80a7e8f"
      + rule_id     = (known after apply)
      + type        = "zone"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

```

**Comment by powellquiring** [comment]
Maybe I'm using the wrong attribute.  I used the "id" instead of the "custom_resolver_id"

```
resource "ibm_dns_custom_resolver" "location" {
  name                      = var.basename
  instance_id               = ibm_resource_instance.dns.guid
  dynamic "locations" {
    for_each = ibm_is_subnet.bug
    content {
        subnet_crn  = locations.value.crn
        enabled     = true
    }
  }
}
```
Then:
```
➜  bug-dns-custom-resolver git:(master) ✗ tf console
> ibm_dns_custom_resolver.location
{
  "created_on" = tostring(null)
  "custom_resolver_id" = "ad005d0a-02e0-4e8f-a231-ba20820971f1"
  "description" = ""
  "enabled" = true
  "health" = "HEALTHY"
  "id" = "ad005d0a-02e0-4e8f-a231-ba20820971f1:8ad2e587-f0a5-4885-87ee-010bca9c1852"
  "instance_id" = "8ad2e587-f0a5-4885-87ee-010bca9c1852"
  "locations" = toset([
```

Here: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/dns_custom_resolver  The problem is missing and wrong docs:

![image](https://user-images.githubusercontent.com/6932057/129776308-5614741c-b629-478c-a89b-1c1ef8cff710.png)


id - (String) The unique ID of the private DNS custom resolver.

A lot of resources use .id as the terraform reference.  You should consider doing the same.  To put together this example notice the use of ...group.id, ibm_is_vpc.bug.id, ...
```
resource "ibm_is_vpc" "bug" {
  name = var.basename
  resource_group = data.ibm_resource_group.group.id
}

resource "ibm_is_subnet" "bug" {
  for_each = toset(local.zones)
  name                     = var.basename
  vpc                      = ibm_is_vpc.bug.id
  resource_group = data.ibm_resource_group.group.id
}

resource "ibm_resource_instance" "dns" {
  name              = var.basename
  resource_group_id = data.ibm_resource_group.group.id
```

If this convention is not followed it is particularly important that the documentation be good and include compete examples demonstrating the resources

Maybe I missed this because there is a bug in the docs.   Notice the missing "ibm_" https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/dns_custom_resolver_forwarding_rule

![image](https://user-images.githubusercontent.com/6932057/129777431-2fb70b2c-359b-4cdf-9a28-e2116c78c0ae.png)

In summary the issue can be resolved by fixing the docs.  Consider following the convention of using ".id" to connect resources.
